### PR TITLE
API Use updated FormField::validate() signature

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/11422

### CODE REVIEW

File: code/Forms/UploadField.php [hunk 1]
```diff
@@ -320,28 +321,25 @@ public function performDisabledTransformation()
...
-            __CLASS__ . '.ErrorMaxFilesReached',
-            'You can only upload {count} file.|You can only upload {count} files.',
-            [
-                'count' => $maxFiles,
-            ]
-        ));
-
-        return $this->extendValidationResult(false, $validator);
+        $result = ValidationResult::create();
+        $this->beforeExtending('updateValidate', function () use ($result) {
```
Comment (ChloeSartorelli) [inline code/Forms/UploadField.php]:
```suggestion
        $this->beforeExtending('updateValidate', function (ValidationResult $result)) {
```

File: code/Forms/UploadField.php [hunk 2]
```diff
@@ -320,28 +321,25 @@ public function performDisabledTransformation()
...
+            }
+            $result->addFieldError($this->getName(), _t(
+                __CLASS__ . '.ErrorMaxFilesReached',
+                'You can only upload {count} file.|You can only upload {count} files.',
+                [
+                    'count' => $maxFiles,
+                ]
+            ));
+        });
+        return $result->combineAnd(parent::validate());
```
Comment (ChloeSartorelli) [inline code/Forms/UploadField.php]:
```suggestion
        return parent::validate();
```

**Review by ChloeSartorelli — CHANGES_REQUESTED** [review]

File: code/Forms/UploadField.php (hunk 2 above)
Comment (emteknetnz) [reply code/Forms/UploadField.php]: Updated

File: code/Forms/UploadField.php (hunk 1 above)
Comment (emteknetnz) [reply code/Forms/UploadField.php]: Updated

**Review by ChloeSartorelli — APPROVED** [review]
LGTM, will merge when all PRs are ready

**Review by ChloeSartorelli — APPROVED** [review]
LGTM

**Pull request merged**